### ISSUE TITLE
Syntax errors in gateway templates with go1.14

### ISSUE THREAD

**Issue by syriak** [istio/istio#22366]
**Bug description**
Due to an update to text/template in go1.14, the following templates with incorrect template syntax will fail when using go1.14:
`/install/kubernetes/helm/istio/charts/gateways/templates/rolebindings.yaml`
`/install/kubernetes/helm/istio/charts/gateways/templates/role.yaml`

This line:
`{{- if ($spec.sds) and (eq $spec.sds.enabled true) }}`
should be:
`{{- if and ($spec.sds) (eq $spec.sds.enabled true) }}`
in both files

NOTE: This is the update from the go1.14 release notes: 
```
text/template
The text/template package now correctly reports errors when a parenthesized argument is used as a function. This most commonly shows up in erroneous cases like {{if (eq .F "a") or (eq .F "b")}}. This should be written as {{if or (eq .F "a") (eq .F "b")}}. The erroneous case never worked as expected, and will now be reported with an error can't give argument to non-function.
```

**Comment by howardjohn** [comment]
This is fixed in master and release 1.5 already

**Comment by howardjohn** [comment]
By #22295

**Comment by invertigo** [comment]
Will this ever be fixed in 1.4?  We're currently having to patch those templates to make istio 1.4.6 installable.

**Comment by Kareem21227gg** [comment]
any soklutions on this?
